scripts/js-api-tests.js

```diff
@@ -4706,6 +4706,25 @@ ${path.relative(process.cwd(), input).replace(/\\/g, '/')}:1:2: ERROR: Unexpecte
    └ entry.js ── 25b ─── 25.0%
 `)
   },
+
+  async rewriteRequireToImport({ esbuild, testDir }) {
+    const outfile = path.join(testDir, 'out.mjs');
+    await esbuild.build({
+      stdin: {
+        contents: `const { extname } = require("path"); export { extname };`,
+      },
+      bundle: true,
+      bundle: true,
+      platform: 'node',
+      format: 'esm',
+      outfile
+    })
+
+    // This needs to use relative paths to avoid breaking on Windows.
+    // Importing by absolute path doesn't work on Windows in node.
+    const result = await import('./' + path.relative(__dirname, outfile))
```

Comment:
Nitpick: have you considered using `url.pathToFileURL`? Something along the following lines:

```suggestion
    const result = await import(url.pathToFileURL(outfile))
```

Quoting from https://github.com/nodejs/node/issues/34765:

> This behavior isn't restricted to windows paths, the same is true for absolute Unix paths as well. The module import specifiers only support URLs, not file paths. To import file paths, they need to be converted to URLs first on all platforms (like you mentioned).
>
> The general problem with supporting both URLs and file paths is that some URLs are also valid (but different!) file paths and vice versa. We decided to go with URLs over file paths to be consistent with other runtimes (e.g. browsers) that are also using URL-based import specifiers.